pallets/nfts/src/tests.rs

```diff
@@ -1954,49 +1954,110 @@ fn approval_lifecycle_works() {
 }
 
 #[test]
-fn check_approval_without_deadline_works() {
+fn check_approval_works_without_deadline_works() {
 	new_test_ext().execute_with(|| {
 		let collection_id = 0;
-		let item_id = 42;
 		let collection_owner = account(1);
+		let item_id = 42;
 		let item_owner = account(2);
 		let delegate = account(3);
 
 		Balances::make_free_balance_be(&item_owner, 100);
+		// TODO: we should return a clear error for smart contracts.
+		// assert_noop!(Nfts::check_collection_approval(&collection_id, &None, &delegate),
```

Comment:
Not very clear to me what is the action plan for this commented out code. Could you explain more?

Reply:
I think we should add a check and error if the collection doesn't exist.